### ISSUE TITLE
Rebuild the app with Swift2.0

### ISSUE THREAD

**Issue by manas15** [zulip/zulip-ios#16]
Hey @lfranchi 

I'm a Computer Software Engineering Student at Delhi Technological University. I want rebuild this application with Swift2.0 as a part of GSOC. It would be great if you can guide me on how to get started with that.

Cheers
Manas Sharma

**Comment by manas15** [comment]
Also I was tried running the objective C version but I got build failed error. And the reason for a failed build is that BrowserViewController files are missing in the source code. Please guide me how to go about it.

**Comment by Dwar3xwar** [comment]
Hey, I'm also looking to contribute as part of GSOC.

Looks like the project is missing a couple of dependecies such as [this one](https://github.com/nbuggia/Browser-View-Controller--iPhone-). However, that project is really old and still uses manual release for memory. Therefore current editions of XCode, which now uses ARC, will not run this project.

Seems like they really need a new Swift project. ¯\_(ツ)_/¯

**Comment by manas15** [comment]
@Dwar3xwar you are right on this one. We need to build this project all over again. I'm also waiting to discuss the same with our mentors.

**Comment by timabbott** [comment]
The iOS app builds -- you just need to be sure to follow the instructions in README.md to fetch the dependencies before building it (BrowserViewController is included via a git submodule).

I should clarify that the app does work and people use it -- we posted the idea of rewriting it in Swift because the app was written in 2013 and the best way to write an iOS app has changed pretty significantly since then, but improving the existing app (rather than rewriting it) is certainly a valid strategy we could choose to take instead.  And it might actually make sense to do a Swift rewrite incrementally regardless.

**Comment by web-mi** [comment]
Hi,

i already started working on a AngularJS app with phonegap because i don't think a native app for each platform is needed and brings a huge workload for a open source project.
Because it's my first app i already tried to upload the app to the store and on windows phone (haven't tried if it works currently because i have no phone available) and android its already available. iOS is in progress but as i read it needs up to 14 days to get verified.
You could find the code here (https://github.com/web-mi/Zulip-Messenger).
I will progress on the app because its my first app and some things have to be fixed.
Some constructive feedback would be great!

Cheers
Marc

**Comment by timabbott** [comment]
I definitely think it's worth thinking seriously about writing the iOS app as a React Native app which we could then also use for Windows Phone and maybe in the future replace the Android app as well.  I haven't heard as much enthusiasm for the PhoneGap stack (generally a lot of folks have performance concerns with that stack), but I would definitely be interested in proposals for doing the iOS app in javascript using React Native.
